From cd2cdcf61b91470f24062bf5355c479c4b45bc45 Mon Sep 17 00:00:00 2001
From: Aravindh Puthiyaparambil <aravindh@redhat.com>
Date: Wed, 17 May 2023 09:48:48 -0700
Subject: [PATCH] UPSTREAM: <carry>: Extend NodeLogQuery feature

Fix handling of the "until" parameter when generating the journalctl
command. This was incorrectly being passed with the "since" value.
---
 pkg/kubelet/kubelet_server_journal_linux.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/kubelet/kubelet_server_journal_linux.go b/pkg/kubelet/kubelet_server_journal_linux.go
index 141337e08f1..bbf0e977226 100644
--- a/pkg/kubelet/kubelet_server_journal_linux.go
+++ b/pkg/kubelet/kubelet_server_journal_linux.go
@@ -40,7 +40,7 @@ func getLoggingCmd(n *nodeLogQuery, services []string) (string, []string, error)
 	}
 
 	if len(n.Until) > 0 {
-		args = append(args, fmt.Sprintf("--since=%s", n.Since))
+		args = append(args, fmt.Sprintf("--until=%s", n.Until))
 	} else if n.UntilTime != nil {
 		args = append(args, fmt.Sprintf("--until=%s", n.SinceTime.Format(dateLayout)))
 	}
-- 
2.40.1

